extensions/parquet/base/src/main/java/io/deephaven/parquet/base/RowGroupReaderImpl.java

```diff
@@ -68,16 +78,29 @@ final class RowGroupReaderImpl implements RowGroupReader {
                     nonRequiredFields.add(fieldType);
                 }
             }
-            chunkMap.put(key, column);
-            schemaMap.put(key, nonRequiredFields);
+            if (chunkMap.putIfAbsent(key, column) != null) {
+                nonUniqueKeys.add(key);
+            }
+            schemaMap.putIfAbsent(key, nonRequiredFields);
             if (ft.getId() != null) {
-                chunkMapByFieldId.put(ft.getId().intValue(), column);
-                schemaMapByFieldId.put(ft.getId().intValue(), nonRequiredFields);
+                final int fieldId = ft.getId().intValue();
+                if (chunkMapByFieldId.putIfAbsent(fieldId, column) != null) {
+                    nonUniqueFieldIds.add(fieldId);
+                }
+                schemaMapByFieldId.putIfAbsent(fieldId, nonRequiredFields);
             }
         }
         if (fieldsIt.hasNext() || colsIt.hasNext()) {
             throw new IllegalStateException(String.format("Unexpected, iterators not exhausted, rootURI=%s", rootURI));
         }
+        for (String nonUniqueKey : nonUniqueKeys) {
+            chunkMap.remove(nonUniqueKey);
```

Comment:
Can you please explain and maybe add a comment for what the behavior will be if we find columns with same field ID?
I thought you were planning to throw an exception here but looks like we will just remove the column mapping?

Reply:
This is for reading, not inference. The logic you are referring to is in inference. I made a comment that the way the code is structured is very bad and leads to this sort of semi-duplicated logic.